core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2881,11 +2883,15 @@ struct ViewValueFunctor<ExecSpace, ValueType, false /* is_scalar */> {
       }
 #endif
       const Kokkos::Impl::ParallelFor<ViewValueFunctor, PolicyType> closure(
-          *this, PolicyType(0, n));
+          *this, policy);
       closure.execute();
       space.fence();
       if (Kokkos::Profiling::profileLibraryLoaded()) {
-        Kokkos::Profiling::endParallelFor(kpID);
+        auto functor_name =
+            (destroy ? "Kokkos::View::destruction [" + name + "]"
+                     : "Kokkos::View::initialization [" + name + "]");
```

Comment:
Did you consider "reusing"  `functor_name` from line 2873?

Reply:
no I don't want the string construction happen if you are not inside the "Do I have profiling on" block. I mean I guess we could explicitly create a std::string before entering the blocks, and then assign inthe first and reuse in the second.

Reply:
Something like `std::optional`.  Anyway David answer my question by saying that was consistent with what he did everywhere else.